Run reconfigurer thread even when restartOnRedeploy is true

bratseth opened the pull request:
Separate system internal redeploys from	application package
changes	such that we can ignore	config updates originating
from application package changes if restart on deploy is true,
as the config change will be followed by a restart - but continue
to effect config changes originating from internal system changes
as those will not (and should not) be followed by a restart.

@gjoranv please review the container/CI parts of this.
@hmusum please review the config parts.

I believe this is complete now but I want to figure out how to test as much as possible of it before merging.

Commit:
Run reconfigurer thread even when restartOnRedeploy is true

gjoranv commented on container-core/src/main/java/com/yahoo/container/core/config/HandlersConfigurerDi.java:
```diff
@@ -138,11 +138,18 @@ public void useBundles(Collection<FileReference> bundles) {
+     *
+     * @return true if this resulted in a new graph that should be applied to the currently running container
+     */
+    public boolean getNewConfigGraph(Injector discInjector, boolean restartOnRedeploy) throws InterruptedException {
```
You probably mean ComponentGraph instead of ConfigGraph

bratseth replied:
Hm, yeah but I also wanted to say "get config" ... but ok I'll rename.

gjoranv reviewed (approved):
Merge after renaming method.

Commit:
Separate internal redeploys from application package changes

Separate system internal redeploys from application package
changes such that we can ignore config updates originating
from application package changes if restart on deploy is true,
as the config change will be followed by a restart - but continue
to effect config changes originating from internal system changes
as those will not (and should not) be followed by a restart.

Commit:
Merge with master

Commit:
Correct JavaDoc

Commit:
Internal redeploy Depoyer.deployFromLocalActive -> ApplicationMetaData

Commit:
Test internalRedeploy propagation in config responses

Commit:
Test internalRedeploy from ApplicationMetaData -> ConfigResponse

Commit:
Test no config when restart on redeploy

Commit:
Remove println

Commit:
Test internalRedepoy persistence
